rbm_softmax.py

```diff
@@ -0,0 +1,504 @@
+"""Restricted Boltzmann Machine with softmax visible units.
```

Comment:
I'm actually going to push the corresponding sklearn file to master, so it's clear which parts here I added, and what's cannibalized (a lot).